dataservice/api/family_relationship/models.py

```diff
@@ -35,62 +35,101 @@ class FamilyRelationship(db.Model, Base):
     """
     __tablename__ = 'family_relationship'
     __prefix__ = 'FR'
-    __table_args__ = (db.UniqueConstraint('participant_id', 'relative_id',
-                                          'participant_to_relative_relation',
-                                          'relative_to_participant_relation'),)
+    __table_args__ = (db.UniqueConstraint(
+        'participant1_id', 'participant2_id',
+        'participant1_to_participant2_relation',
+        'participant2_to_participant1_relation'),)
     external_id = db.Column(db.Text(),
                             doc='external id used by contributor')
-    participant_id = db.Column(
+    participant1_id = db.Column(
         KfId(),
         db.ForeignKey('participant.kf_id'),
         nullable=False,
         doc='kf_id of one participant in the relationship')
 
-    relative_id = db.Column(
+    participant2_id = db.Column(
         KfId(),
         db.ForeignKey('participant.kf_id'),
         nullable=False,
         doc='kf_id of the other participant in the relationship')
 
-    participant_to_relative_relation = db.Column(db.Text(), nullable=False)
+    participant1_to_participant2_relation = db.Column(db.Text(),
+                                                      nullable=False)
 
-    relative_to_participant_relation = db.Column(db.Text())
+    participant2_to_participant1_relation = db.Column(db.Text())
 
-    participant = db.relationship(
+    participant1 = db.relationship(
         Participant,
-        primaryjoin=participant_id == Participant.kf_id,
+        primaryjoin=participant1_id == Participant.kf_id,
         backref=db.backref('outgoing_family_relationships',
                            cascade='all, delete-orphan'))
 
-    relative = db.relationship(
+    participant2 = db.relationship(
         Participant,
-        primaryjoin=relative_id == Participant.kf_id,
+        primaryjoin=participant2_id == Participant.kf_id,
         backref=db.backref('incoming_family_relationships',
                            cascade='all, delete-orphan'))
 
     @classmethod
-    def query_all_relationships(cls, participant_kf_id):
-        """
-        Find all relationship records given a the kf_id of a particpant
+    def query_all_relationships(cls, participant_kf_id=None,
```

Comment:
It would be nice if we could figure out how to turn this into an extension of Query only for FamilyRelationships.
Then we could do:
```
FamilyRelationships.query.filter_by(foo=bar).all_relationships(participant_id='blah').count()
```